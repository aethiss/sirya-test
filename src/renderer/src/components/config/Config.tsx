import { ReactElement } from 'react'

interface Props {
  salt?: string
}

const Config = ({ salt }: Props): ReactElement => {
  const handleConfig = (): void => {
    window.electron.ipcRenderer.send('open-file-dialog')
  }

  return (
    <div>
      <button data-testid="config-button" onClick={handleConfig}>
        {salt ? 'Update config file' : 'Select config file'}
      </button>
    </div>
  )
}

export default Config
