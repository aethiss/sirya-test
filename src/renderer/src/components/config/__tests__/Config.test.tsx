/* eslint-disable @typescript-eslint/ban-ts-comment */
import '@testing-library/jest-dom'
import { render, fireEvent } from '@testing-library/react'

import Config from '../Config'

// Mocks
// jest.mock('electron', () => ({
//   ipcRenderer: {
//     send: jest.fn()
//   }
// }))

describe('Config', () => {
  describe('Rendering', () => {
    test('Renders the config page without file', () => {
      const { asFragment, getByTestId } = render(<Config />)
      expect(asFragment()).toMatchSnapshot()
      const configButton = getByTestId('config-button')
      expect(configButton).toBeInTheDocument()
      expect(configButton).toHaveTextContent('Select config file')
    })

    test('Renders the config page with file', () => {
      const { getByTestId } = render(<Config salt="salt" />)
      const configButton = getByTestId('config-button')
      expect(configButton).toHaveTextContent('Update config file')
    })
  })

  describe('Functionality', () => {
    beforeAll(() => {
      // @ts-ignore
      window.electron = { ipcRenderer: { send: jest.fn() } }
    })

    test('test if open dialog', async () => {
      const { getByTestId } = render(<Config />)
      const configButton = getByTestId('config-button')
      fireEvent.click(configButton)
      console.log('win', JSON.stringify(window))
      expect(window.electron.ipcRenderer.send).toHaveBeenCalledTimes(1)
    })
  })
})
