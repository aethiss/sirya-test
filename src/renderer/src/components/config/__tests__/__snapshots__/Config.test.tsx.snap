// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config Rendering Renders the config page without file 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="config-button"
    >
      Select config file
    </button>
  </div>
</DocumentFragment>
`;
